src/discovery/HazelcastCloudDiscovery.ts

```diff
@@ -73,6 +73,10 @@ export class HazelcastCloudDiscovery {
 
         let dataAsAString = '';
         const req = get(endpointUrlOptions, (res: IncomingMessage) => {
+            if (res.statusCode != 200) {
+                deferred.reject(new HazelcastError('Your cluster discovery token is invalid.'));
+                return;
```

Comment:
what data is returned as a response if the token is invalid? Also can you add a test testing your change?  You can use the config from the github issue: `{ network: { hazelcastCloud: { discoveryToken: 'YOUR_DISCOVERY_TOKEN' } }, clusterName: 'YOUR_CLUSTER_NAME' }`

also you can also use some other random `discoveryToken`s for test variety. 

I could not find a integration test file (which are under `test/integration`) about cloud discovery, but you can create a new file. Alternatively, you can write a unit test by mocking the response.

Reply:
What should be returned?

Reply:
Two things I wanted here:

1. I wonder what is returned from cloud response when token is invalid. Can you test it to learn?
2. Can you add a unit test testing the logic you added, i.e your new if block. You can use a random token and assert that it throws a error with proper message (`Your cluster discovery token is invalid`). Instead of sending a real request to cloud in the test, you can mock sending a request using sinon.js. If that is too complex for you, let's send a real request.

Reply:
I have learned 1 by myself. Returned response 

```
Caused by: java.io.IOException: {"application":"orchestrator-service","message":"Cluster token not found for token.","status":404,"error":"Not Found","errorCode":"ClusterTokenNotFound","timestamp":1663100479625,"path":"https://coordinator.hazelcast.cloud/cluster/discovery","method":"GET"}
	at com.hazelcast.client.impl.spi.impl.discovery.HazelcastCloudDiscovery.checkError(HazelcastCloudDiscovery.java:141)
	at com.hazelcast.client.impl.spi.impl.discovery.HazelcastCloudDiscovery.callService(HazelcastCloudDiscovery.java:85)
	at com.hazelcast.client.impl.spi.impl.discovery.HazelcastCloudDiscovery.discoverNodes(HazelcastCloudDiscovery.java:70)
	... 10 more

```